docs/profile-data.md

```diff
@@ -0,0 +1,236 @@
+
+```js
+{
+  // JSON string of the executable library data. This needs to be JSON parsed a
+  // second time.
+
+  libs: `[{
+    {
+      "start": 4461400064,
+      "end": 4461424640,
+      "offset": 0,
+      "name": "/Applications/FirefoxNightly.app/Contents/MacOS/firefox",
+      "breakpadId": "E54D3AF274383256B9F6144F83F3F7510"
+    },
+    ...
+  }]`,
+
+  // Key/pair meta values about the profile.
+
+  meta: {
+    version: 3,
+    interval: 0.4,
+    stackwalk: 1,
+    startTime: 1477063882018.4387,
+    processType: 0,
+    platform: "Macintosh",
+    oscpu: "Intel Mac OS X 10.12",
+    misc: "rv:52.0",
+    abi: "x86_64-gcc3",
+    toolkit: "cocoa",
+    product: "Firefox"
+  },
+
+  // All of the threads that were profiled.
+
+  threads: [
+    {
+      //--------------------------------------------------------
+      // Profile data
+      //--------------------------------------------------------
+
+      // Markers represents events within the browser. These can be a wide variety of
+      // events from different systems within the browser. For instance these can
+      // include GC pauses, Paints, Reflows, DOM events, etc.
+
+      markers: [
+        SchemaData({
+          name // index into stringTable - Marker name
+          time // TODO e.g. 37070.186708
+          data // arbitrary JSON about the marker. e.g.
+               // {
+               //   "type": "tracing",
+               //   "category": "Paint",
+               //   "interval": "start"
+               // }
+        }),
+        ...
+      ],
+
+      // The profiler regularly samples the current frame on the call stack. These
+      // samples provide a picture into what is executing through the time of the
+      // profile. These can be C++ frames or JS frames.
+
+      samples: [
+        SchemaData({
+          stack,           // index into stackTable - The current stack of the sample.
+          time,            // TODO - unit? The time of the sample: 37067.409299
+          responsiveness,  // TODO Length of the time for the sample? e.g. 1.437998
+          rss,             // TODO
+          uss,             // TODO
+          frameNumber,     // TODO
+          power            // TODO
+        }),
+        ...
+      ],
+
+      //--------------------------------------------------------
+      // Misc properties
+      //--------------------------------------------------------
+
+      // The thread ID - TODO: how is this useful?
+      tid: 7442229,
+
+      // The name of this thread. - TODO: what are the possible values here?
+      name, // String e.g. "GeckoMain", "Compositor", etc.
```

Comment:
This is the name that was passed to Sampler::RegisterCurrentThread on that thread.